jujugui/static/gui/src/app/mega-watcher/mega-watcher.js

```diff
@@ -0,0 +1,80 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const deepmerge = require('deepmerge');
+
+class MegaWatcher {
+  constructor(config) {
+    this.ALL_WATCHER_EVENT = config.changeEvent;
+    this.ON_CHANGE = config.onChange;
+    this._boundWatcherListener = this._watcherListener.bind(this);
+    this.entities = {
+      annotations: {},
+      applications: {},
+      machines: {},
+      relations: {},
+      'remote-applications': {},
+      units: {}
+    };
+  }
+
+  getEntities() {
+    return JSON.parse(JSON.stringify(this.entities));
+  }
+
+  connect() {
+    document.addEventListener(this.ALL_WATCHER_EVENT, this._boundWatcherListener);
```

Comment:
Instead of relying on global events I feel like the megawatcher should take a readable stream and then possibly provide a writable stream out (if we choose to do any data munging).